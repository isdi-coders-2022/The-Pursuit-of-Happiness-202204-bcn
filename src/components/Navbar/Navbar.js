import styled from "styled-components";

const StyledNavbarComponenent = styled.div`
  color: black;
  display: flex;
  h2 {
    font-size: 22px;
    color: white;
  }
`;

const Navbar = ({ homeText, favText, className }) => {
  return (
    <StyledNavbarComponenent>
      <h2 className={className}>{homeText}</h2>
      <h2 className={className}>{favText}</h2>
    </StyledNavbarComponenent>
  );
};

export default Navbar;
