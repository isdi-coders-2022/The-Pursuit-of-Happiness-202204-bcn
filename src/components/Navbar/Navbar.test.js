import { BrowserRouter } from "react-router-dom";
import TestRenderer from "react-test-renderer";
import Navbar from "./Navbar";
import { render, screen } from "@testing-library/react";

describe("Given Navbar function", () => {
  describe("When it's called with to word given", () => {
    test("Then it should render two headings", () => {
      const textHome = "Home";
      const textFavorites = "Favorites";

      render(
        <BrowserRouter>
          <Navbar homeText={textHome} favText={textFavorites} />
        </BrowserRouter>
      );

      const expectedNavbarText = screen.getAllByRole("heading");

      expect(expectedNavbarText).toHaveLength(2);
    });
  });
  test("Then it should always match for this snapshot", () => {
    const navbarReceived = TestRenderer.create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(navbarReceived).toMatchSnapshot();
  });
});
