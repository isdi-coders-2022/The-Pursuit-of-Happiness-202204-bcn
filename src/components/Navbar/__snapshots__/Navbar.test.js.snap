// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given Navbar function When it's called with to word given Then it should always match for this snapshot 1`] = `
<div
  className="sc-bczRLJ evcdSO col-8"
>
  <a
    className=""
    href="/HomePage"
    onClick={[Function]}
  >
    <h2
      className="col-6"
    />
  </a>
  <a
    className=""
    href="/FavsPage"
    onClick={[Function]}
  >
    <h2
      className="col-6"
    />
  </a>
</div>
`;
